sdk/core/Azure.Core/src/Shared/TaskExtensions.cs

```diff
@@ -87,5 +93,39 @@ private static void VerifyTaskCompleted(bool isCompleted)
                 throw new InvalidOperationException("Task is not completed");
             }
         }
+
+        public readonly struct Enumerable<T> : IEnumerable<T>
+        {
+            private readonly IAsyncEnumerable<T> _asyncEnumerable;
+
+            public Enumerable(IAsyncEnumerable<T> asyncEnumerable) => _asyncEnumerable = asyncEnumerable;
+
+            public Enumerator<T> GetEnumerator() => new Enumerator<T>(_asyncEnumerable.GetAsyncEnumerator());
+
+            IEnumerator<T> IEnumerable<T>.GetEnumerator() => new Enumerator<T>(_asyncEnumerable.GetAsyncEnumerator());
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        public readonly struct Enumerator<T> : IEnumerator<T>
+        {
+            private readonly IAsyncEnumerator<T> _asyncEnumerator;
+
+            public Enumerator(IAsyncEnumerator<T> asyncEnumerator) => _asyncEnumerator = asyncEnumerator;
+
+#pragma warning disable AZC0107 // Do not call public asynchronous method in synchronous scope.
+            public bool MoveNext() => _asyncEnumerator.MoveNextAsync().EnsureCompleted();
+#pragma warning restore AZC0107 // Do not call public asynchronous method in synchronous scope.
+
+            public void Reset() => throw new NotSupportedException();
```

Comment:
We should put a helpful message in this for the poor developer who tries to do this someday.  Maybe something like `$"The {nameof(IAsyncEnumerator<T>)} wrapped by this {nameof(IEnumerator<T>)} cannot be {nameof(Reset)}!`?

Reply:
Fixed.